merlin/parser.py

```diff
@@ -0,0 +1,36 @@
+import yaml
+
+from merlin.metric import SourceMetric, Definition
+from merlin.stage import Stage
+from merlin.utils import LOGGER
+
+
+class MetricParser():
+    def __init__(self):
+        # the metric_fields is a set of metric field strings. This is for validation use.
+        self.metric_fields = set(['id', 'time', 'w_time', 'group_map', 'group_keys', 'func_map', 'func_expr', 'func_vars', 'compute_time', 'v_lvl', 'h_lvl', 'stages', 'description', 'version'])
```

Comment:
I think you can figure out the fields from SoruceMetric... 
In any case `group_map` `group_keys` `func_map` will not be part of the source metric or definition